workers/unity/Packages/com.improbable.gdk.core/Systems/CleanReactiveComponentsSystem.cs

```diff
@@ -54,32 +53,40 @@ private void GenerateComponentGroups()
                         continue;
                     }
 
-                    if (typesToRemove.Contains(type))
+                    if (typesToRemove.Add(type))
                     {
-                        continue;
+                        addRemoveComponentActionMethod.MakeGenericMethod(type).Invoke(this, null);
                     }
-
-                    typesToRemove.Add(type);
-                    addRemoveComponentActionMethod.MakeGenericMethod(type).Invoke(this, null);
                 }
 
                 // Find all ComponentCleanupHandlers
                 foreach (var type in assembly.GetTypes())
                 {
                     if (typeof(ComponentCleanupHandler).IsAssignableFrom(type) && !type.IsAbstract)
                     {
+                        // These are generic and cannot use `RemoveAtEndOfTick`
```

Comment:
Remove at end of tick doesn't have a problem with generic stuff, the problem was previously the pooling, and is currently the freeing of memory.

Reply:
Should add that it doesn't have a problem with generics in the case where the generic constraint can be used to work out all concrete types with no extras. Provided this works for IL2CPP.